File: 0380-insert-delete-getrandom-o1/0380-insert-delete-getrandom-o1.py
import random
class RandomizedSet:

    def __init__(self):
        
        self.store = []

    def insert(self, val: int) -> bool:
        if val in self.store:
            return False
        else:
            self.store.append(val)
            return True
    
    def remove(self, val: int) -> bool:
        if val in self.store:
            self.store.remove(val)
            return True
        else:
            return False
        

    def getRandom(self) -> int:
        return random.choice(self.store)
        


# Your RandomizedSet object will be instantiated and called as such:
# obj = RandomizedSet()
# param_1 = obj.insert(val)
# param_2 = obj.remove(val)
# param_3 = obj.getRandom()

File: 0380-insert-delete-getrandom-o1/0380-insert-delete-getrandom-o1.scala
import scala.collection.mutable
import scala.collection.mutable.ArrayBuffer
import scala.util.Random
class RandomizedSet() {

    private val dict = mutable.HashMap[Int, Int]()
    private val list = ArrayBuffer[Int]()


    def insert(`val`: Int): Boolean = {
      if(dict.contains(`val`)){
        false
      }
      else{
        dict(`val`) = list.size
        list.append(`val`)
        true
      }
    }

    def remove(`val`: Int): Boolean = {

      if(!dict.contains(`val`)){
        false
      }
      else{
        val idx = dict(`val`)
        val lastElement = list.last
        list(idx) = lastElement
        dict(lastElement) =  idx
        list.dropRightInPlace(1)
        dict.remove(`val`)
        true
      }

    }

    def getRandom(): Int = {
      list(Random.nextInt(list.size))

    }
}

/**
 * Your RandomizedSet object will be instantiated and called as such:
 * val obj = new RandomizedSet()
 * val param_1 = obj.insert(`val`)
 * val param_2 = obj.remove(`val`)
 * val param_3 = obj.getRandom()
 */